---
title: "Solution 4"
author: "Hubert Rehrauer"
date: "8 10 2017"
output: html_document
---



## Mapping in practice

Check the online available manuals for the mappers

* STAR
* bowtie2
* tophat


<!-- ### STAR scoring and reporting -->
<!-- The most importatn info is not in the manual: a match gives +1, a mismatch -1. All other options are described in the section "Scoring" in the STAR manual. -->

<!-- STAR is very lenient and tolerates a large fraction of mismatches.  -->

<!-- * bowtie2   -->
<!-- * tophat2   -->

and answer the questions  
* How do these mappers score alignments?  
* What reporting options are available?  


## Mapping with R

Use the Rsubread package to map a read file found at http://fgcz-gstore.uzh.ch/public/Yeast_Reads/


```{r, eval=FALSE}
source("https://bioconductor.org/biocLite.R")
biocLite("Rsubread")
biocLite("BSgenome.Scerevisiae.UCSC.sacCer3")
biocLite("TxDb.Scerevisiae.UCSC.sacCer3.sgdGene")
```



```{r "setup", include=FALSE}
dir.create("~/tmp/Yeast_example")
setwd("~/tmp/Yeast_example")
require("knitr")
opts_chunk$set(echo = TRUE)
opts_knit$set(root.dir = "~/tmp/Yeast_example")
library("BSgenome.Scerevisiae.UCSC.sacCer3")
library("Rsubread")
```

Use the commands *export* to export the genome as a fasta file. Use the commands *buildindex* and *align* to map the reads. Consult the help pages of the respective functions. Build the index so that at most 2GB of RAM is used. Run the alignment with 4 threads. If you computer has less resources, adapt accordingly.

```{r}
export(BSgenome.Scerevisiae.UCSC.sacCer3, con="sacCer3-genome.fasta")
buildindex("sacCer3-Rsub", "sacCer3-genome.fasta", memory=2000 )
```

```{r}
align(index = "sacCer3-Rsub", "20170313.A-G1_R1-sub.fastq.gz", output_file="mapped.bam", nthreads=4)
```


```{r}
align(index = "sacCer3-Rsub", "20170313.A-G1_R1-sub.fastq.gz", output_file="mapped.bam", nthreads=4, nTrim5 = 5, nTrim3 = 5)
```


What is the mapping rate of the reads? How does the mapping rate change if you trim 5 bases of both ends?



